crates/merod/src/version.rs

```diff
@@ -0,0 +1,28 @@
+use eyre::Result as EyreResult;
+use reqwest::Client;
+use semver::Version;
+use serde::Deserialize;
+
+const CURRENT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+pub async fn check_for_update(client: &Client) -> EyreResult<()> {
+    let url = "https://api.github.com/repos/calimero-network/core/releases/latest";
+
+    let response = client
+        .get(url)
+        .header("User-Agent", "merod-version-check")
+        .send()
+        .await?;
+    #[derive(Deserialize)]
+    struct Release {
+        tag_name: String,
+    }
+
+    let release: Release = response.json().await?;
+    let latest = Version::parse(&release.tag_name)?;
+    let current = Version::parse(CURRENT_VERSION)?;
```

Comment:
why not use `Version` in the struct directly? (you'd need the `serde` feature flag for the `semver` crate)

```suggestion
    #[derive(Deserialize)]
    struct Release {
        tag_name: Version,
    }

    let release: Release = response.json().await?;
```

and as for `CURRENT_VERSION`, use a static variable instead of the const (see [`LazyLock`](https://doc.rust-lang.org/std/sync/struct.LazyLock.html))